### PR TITLE
[FAB-17135] Private data sharing doc

### PR DESCRIPTION
**Pull request by denyeart**
#### Type of change

- Documentation update

#### Description

Document private data enhancements in v2.0 including
private data sharing patterns and example.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

### CODE REVIEW

File: docs/source/private-data/private-data.md [hunk 1]
```diff
@@ -140,6 +143,184 @@ documentation on [transaction flow](../txflow.html).
...
 
+## Sharing private data
+
+While you can create sets of multilateral private data collections to share data
+among collection members, this approach may result in a large number of collections
+that need to be defined. It also does not address the need to share or transfer
+private data beyond the collection members at the transaction or key level.
+For these reasons, consider using a private data collection per organization,
+and then sharing private data with other organization-specific collections as the
+need arises. Starting in Fabric v2.0, implicit organization-specific collections are
```
Comment (mastersingh24) [inline docs/source/private-data/private-data.md]: Are implicit organization-specific collections described somewhere?  I might have missed it in the existing docs, but I did not see anything describing how to use/access them.

File: docs/source/private-data/private-data.md [hunk 2]
```diff
@@ -140,6 +143,184 @@ documentation on [transaction flow](../txflow.html).
...
+  channel members interested in the asset may query the history of the public
+  key to understand its provenance, but will not have access to any private
+  details unless an owner shares it on a need-to-know basis.
+
+The basic asset transfer scenario could be extended for other considerations,
+for example the transfer chaincode could verify that a payment record is available
+to satisfy payment versus delivery requirements, or verify that a bank has
+submitted a letter of credit, prior to the execution of the transfer chaincode.
+And instead of clients directly hosting peers, they could transact through custodian
+organizations.
```
Comment (mastersingh24) [inline docs/source/private-data/private-data.md]: I think we will need some type of enhanced diagram here .... you might also want to think about actually creating a full-fledged example of this ... although that will take some thought.

File: docs/source/private-data/private-data.md (hunk 1 above)
Comment (denyeart) [reply docs/source/private-data/private-data.md]: I'll be using the same Jira issue to create reference doc for each of the v2.0 enhancements.

File: docs/source/private-data/private-data.md (hunk 2 above)
Comment (denyeart) [reply docs/source/private-data/private-data.md]: This is actually a teaser and more generic version of the Commercial Paper Advanced sample that Anthony was working on. @ODOWDAIBM do you still intend to complete Commercial Paper Advanced sample in v2.0 timeframe?

**Review by mastersingh24 — DISMISSED** [review]

**Review by pamandrejko — COMMENTED** [review]
Part of me feels like some of this content (how-to instructions?) is too low level for the key concepts topic, but I'm not sure how-to instructions go in the architecture topic either.  We don't currently have a topic on private data in the Operations guide.

**Comment by denyeart** [comment]
> Part of me feels like some of this content (how-to instructions?) is too low level for the key concepts topic, but I'm not sure how-to instructions go in the architecture topic either. We don't currently have a topic on private data in the Operations guide.

Yeah, I thought it was most important to get the content in now so that we can highlight for v2.0 beta, and then decide how to refactor the topics for best structure.

**Comment by denyeart** [comment]
@pamandrejko All comments addressed, I'll go ahead and force merge, we can revisit the overall private data structure.

**Pull request merged**